File: 2502-sort-the-people/sort-the-people.cpp
class Solution {
public:
    vector<string> sortPeople(vector<string>& names, vector<int>& heights) {
        vector<pair<string,int>>v;
        for(int i=0;i<names.size();i++){
             v.push_back({names[i],heights[i]});
        }

        sort(v.begin(),v.end(),[&](pair<string,int>a, pair<string,int>b){
            return a.second>b.second;
        });

        for(int i=0;i<names.size();i++){
             names[i] = v[i].first;
        }

        return names;
    }
};

File: 2502-sort-the-people/sort-the-people.java
class Solution {
    public String[] sortPeople(String[] names, int[] heights) {
        List<Integer> nameIndexList = new ArrayList<>();
        for(int i=0;i<names.length;i++){
            nameIndexList.add(i);
        }

        return nameIndexList.stream().
         sorted((a,b)->-(heights[a]-heights[b])).
         map(element->names[element]).
         toArray(String[]::new);
    }
}